### ISSUE TITLE
Errors not showing on the issues panel

### ISSUE THREAD

**Issue by nye** [VineCode/PHPLint.novaextension#6]
When I save a php document with a syntax error (like a missing ;) nothing is showing on the issues panel. 

I can see a message on the extension log: `PHPLint` `Errors parsing /some/project/some/file.php` but nothing on the issues panel.

I gues it is supossed to show someting in the issues panel, right?
Do you know is there's any problem with this?

Thanks

**Comment by oliveratgithub** [comment]
+1 Same here.

On save I get in the debug log:
`Errors parsing /Volumes/macOS/Users/me/Sites/project/index.php`

(when I was testing the fix in v0.3.1 that was release due to #3 )

**Comment by dansroka** [comment]
+1. I am getting the same message in the Extension Console (under logs, not errors or messages).

Don't know if this helps, but I am also noticing an odd behavior with this log message: even though I have my PHP validation set to "On Save" (under Project settings > PHP Environment > Validation), this message gets logged after each change to the file — but only after I save the file with an error a first time.

For example, I am working on this file:
```
<?php
echo "hello";
?>
```
I remove the semicolon in the `echo "hello"` line — I get no message in the console, as expected.
I save the save the file, and I get this message, as expected:
```
PHPLint[12:22:28.191000] Errors parsing /Volumes/Sparky/Users/sroka/Sites/danielsroka/test.php
```
But now, every time I make a change to the document (without fixing the coding error), the same message gets logged with every change to the text in the file.

**Comment by vinecode-andy** [comment]
@dansroka I am looking into this but just for info the settings in Project settings > PHP Environment > Validation are built into Nova rather than being related to the extension. To be completely honest I didn't know it existed :)

**Comment by vinecode-andy** [comment]
@oliveratgithub @dansroka @nye If possible guys could you let me know the PHP versions you are running. In Terminal  `/usr/bin/php --version` and let me know the output. I am running PHP 7.3.11 (cli) and still can't seem to re-produce this.

**Comment by dansroka** [comment]
I'm on PHP 7.3.11.

OK, I was just double checking this issue and figured some things out, I believe.

First, I tested php lint by running it directly from the terminal:

`php -l test.php`

And was getting the same un-helpful error message as the PHPLint extension was giving me in Nova:

`Errors parsing /Volumes/Sparky/Users/sroka/Sites/danielsroka/test.php`

I decided to check my error settings in /etc/php.ini, and found my problems. First, display_errors was set to Off. So I turned it on, but got the same generic "errors parsing..." message. So I then changed my errors_reporting from E_ERROR & E_WARNING & E_PARSE to

`error_reporting =  E_ALL & ~E_NOTICE`

...and that fixed it. The php -l now returns the full error message in the terminal, and when I am in Nova, any error messages from PHPLint are now appearing in the Issues panel.

**Comment by dansroka** [comment]
Also, the "Validation" setting in Nova (On Save, or On Change) is now working as expected. Possibly because I have my error_reporting fixed?

**Comment by vinecode-andy** [comment]
@dansroka Thanks for the update, this is a huge help. I will see if I can get round this or at least document it. 
It does look like the "Validation" setting in Nova just uses the same php -l as I do so it looks to be the same issue. 

Thanks for your feedback and help, much appreciated.

**Comment by dansroka** [comment]
Just feeling stupid that I didn't check php.ini before! Thanks for all your help!

**Comment by vinecode-andy** [comment]
No worries, it was stumping me too. I am going to update the params to ignore the php.ini and set the error reporting to all when running the lint.

**Comment by nye** [comment]
I had the same problem with the php.ini. display_errors was Off.

Thank you all!

**Comment by vinecode-andy** [comment]
@nye Thanks for the feedback

**Comment by cybervanes** [comment]
I've tried everything, I can't get it to work. I've verified proper .ini settings. any updates coming soon?

**Comment by rodenboog** [comment]
After the same issues I managed to show errors "On Save" but not "On Change". Validation is on in Project Settings, PHP.ini is set correctly. PHP version is 7.4.12, nova 3, PHPLint 0.3.1. Am I missing something?

**Comment by vinecode-andy** [comment]
@cybervanes I have an update which will ignore any ini settings and set the correct error reporting. Unfortunately I have not had chance to complete testing but will try to get on it ASAP.

**Comment by frees11** [comment]
Hi, is there any progress? Today I installed Nova free trial, instlalled your extension, but it doesn't display any syntax errors in issues panel, nor anywhere else in the editor. 

Then I found out about this issue, I tried to setup display errors and error reporting settings in php.ini and in macos terminal  after php -l command I am able to see full error message.

Sadly in Nova it still isn't displayed anywhere. Could you please post some video how this functionality should work? I am completely lost. In project settings I found about on change or on save settings, but it doesn't do anything

**Comment by vinecode-andy** [comment]
Thanks for reporting this, and apologies for the delayed response. Yes—syntax errors should appear in the Issues panel, not just the Extension Console.

Version 1.0 reworks error reporting to read PHP’s standard output and error output, then show reported errors in Issues and highlight the relevant lines in the editor.

Save a local PHP file to check it automatically, or focus it and choose Extensions → PHPLint → Lint Now.

Lint on Save and Show Errors are enabled by default. You can find these settings under Project → Project Settings → Languages & Extensions → PHPLint.

Fixing an error and checking again clears it. If PHP itself cannot complete the check, existing diagnostics are retained and the failure is explained separately.

I’m closing this as addressed in version 1.0. If the problem persists after updating, please open a new issue with your Nova version, PHP version and full Extension Console output.
